pkg/systemd/services/service-details.jsx

```diff
@@ -319,6 +332,45 @@ export class ServiceDetails extends React.Component {
                 });
     }
 
+    deleteAction() {
+        // Stop timer so we don't get race conditions when the unit is gone.
+        if (this.interval)
+            clearInterval(this.interval);
+
+        const promises = [];
+        if (this.props.unit.UnitFileState === "enabled") {
+            if (this.props.unit.ActiveState === "active" || this.props.unit.ActiveState === "activating")
+                promises.push(this.unitAction("StopUnit"));
+            if (this.props.unit.ActiveState === "failed")
+                promises.push(this.unitAction("ResetFailedUnit", []));
+            promises.push(this.unitFileAction("DisableUnitFiles", undefined));
+        }
+
+        // Remove unit files
+        Promise.all(promises).then(() => {
+            const deletions = [];
+
+            deletions.push(cockpit.file(this.props.unit.FragmentPath, { superuser: "required" }).replace(null));
+            this.props.unit.Triggers.forEach(trigger => {
+                trigger = trigger.replace('.', '_2e');
+                deletions.push(
+                    systemd_client[this.props.owner].call(`/org/freedesktop/systemd1/unit/${trigger}`,
+                                                          "org.freedesktop.DBus.Properties", "Get",
+                                                          ["org.freedesktop.systemd1.Unit", 'FragmentPath']).then(([dbus_obj]) => {
+                        if (dbus_obj.v.startsWith("/etc/systemd/system")) {
+                            cockpit.file(dbus_obj.v, { superuser: "required" }).replace(null);
```

Comment:
This Should Not Fail™, but let's at least console.error() if it does, to avoid a page crash and make this debuggable.